src/main/scala/mesosphere/marathon/core/storage/store/impl/memory/InMemoryPersistenceStore.scala

```diff
@@ -148,6 +148,26 @@ class InMemoryPersistenceStore(implicit
 
     Future.successful(Done)
   }
+
+  @volatile private[this] var migrationInProgress: Boolean = false
+
+  override def startMigration(): Future[Done] = {
+    require(isOpen, "the store must be opened before it can be used")
+    if (migrationInProgress) {
+      throw new IllegalStateException("Migration is already in progress")
+    }
+    migrationInProgress = true
```

Comment:
I'm afraid it's not very thread-safe. We can fix this using atomic boolean.

Reply:
Initially, this method was doing nothing, because it doesn't store anything in an external storage, so there is not distribution. It is purely for completeness and uniformity just for unit-tests. Though I agree that it makes sense to do it right, since it is just one-two lines more.